docs/getting-started-guides/ubuntu.md

```diff
@@ -0,0 +1,177 @@
+
+     10.0.0.0        -   10.255.255.255  (10/8 prefix)
+
+     172.16.0.0      -   172.31.255.255  (172.16/12 prefix)
+
+     192.168.0.0     -   192.168.255.255 (192.168/16 prefix) 
+
+The `FLANNEL_NET` variable defines the IP range used for flannel overlay network, should not conflict with above PORTAL_NET range
+
+After all the above variable being set correctly. We can use below command in cluster/ directory to bring up the whole cluster.
+
+`$ KUBERNETES_PROVIDER=ubuntu ./kube-up.sh` 
+
+The scripts is automatically scp binaries and config files to all the machines and start the k8s service on them. The only thing you need to do is to type the sudo password when promoted. The current machine name is shown below like. So you will not type in the wrong password.
+
+```
+
+Deploying minion on machine 10.10.103.223
+
+...
+
+[sudo] password to copy files and start minion: 
+
+```
+
+If all things goes right, you will see the below message from console
+`Cluster validation succeeded` indicating the k8s is up.
+
+**All done !**
+
+You can also use kubectl command to see if the newly created k8s is working correctly. 
+
+For example, use `$ kubectl get minions` to see if you get all your minion nodes comming up and ready. It may take some times for the minions be ready to use like below. 
+
+```
+NAME                 LABELS             STATUS
+
+10.10.103.162       <none>              Ready
+
+10.10.103.223       <none>              Ready
+
+10.10.103.250       <none>              Ready
+
+```
+
+Also you can run kubernetes [guest-example](https://github.com/GoogleCloudPlatform/kubernetes/tree/master/examples/guestbook) to build a redis backend cluster on the k8s．
+
+
+#### IV. Deploy addons
+
+After the previous parts, you will have a working k8s cluster, this part will teach you how to deploy addones like dns onto the existing cluster.
+
+The configuration of dns is configured in cluster/ubuntu/config-default.sh.
+
+```
+
+ENABLE_CLUSTER_DNS=true
+
+DNS_SERVER_IP="192.168.3.10"
+
+DNS_DOMAIN="kubernetes.local"
+
+DNS_REPLICAS=1
+
+```
+The `DNS_SERVER_IP` is defining the ip of dns server which must be in the portal_net range.
+
+The `DNS_REPLICAS` describes how many dns pod running in the cluster.
+
+After all the above variable have been set. Just type the below command
+
+```
+
+$ cd cluster/ubuntu
+
+$ KUBERNETES_PROVIDER=ubuntu ./deployAddons.sh
+
+```
+
+After some time, you can use `$ kubectl get pods` to see the dns pod is running in the cluster. Done!
+
+
+#### IV. Trouble Shooting
+
+Generally, what of this approach did is quite simple: 
+
+1. Build and copy binaries and configuration files to proper dirctories on every node
+
+2. Configure `etcd` using IPs based on input from user 
+
+3. Create and start flannel network
+
+So, whenver you have problem, do not blame Kubernetes, **check etcd configuration first** 
```

Comment:
s/whenevr/whenever/
Maybe change to say: If you see a problem, try the following :